pkg/kubelet/certificate/certificate_manager.go

```diff
@@ -276,7 +276,7 @@ func getCurrentCertificateOrBootstrap(
 		return nil, false, fmt.Errorf("unable to parse certificate data: %v", err)
 	}
 	bootstrapCert.Leaf = certs[0]
-	return &bootstrapCert, true, nil
+	return &bootstrapCert, false, nil
```

Comment:
this will remove rotation immediately after upgrade.  but are we sure that rotation on expired certs actually working ?

Reply:
expired certs cause `m.setRotationDeadline()` to set deadline to rotate immediately, which causes immediate background rotation

Reply:
I don't think this is going to work correctly. The bootstrap certificate may be a different user from the certificate after rotation, which could mean different API permissions. And the bootstrap certificate could be really long lived. Returning true here is supposed to cause a long lived bootstrap certificate to be abandoned immediately.

Reply:
bootstrap here is poorly named. this is the cert from `--kubeconfig`, not `--bootstrap-kubeconfig`